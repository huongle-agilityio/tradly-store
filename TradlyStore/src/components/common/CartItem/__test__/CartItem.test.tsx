import { render, screen, fireEvent } from '@testing-library/react-native';

// Components
import { CartItem } from '..';

describe('CartItem', () => {
  const mockOnRemoveItem = jest.fn();
  const mockOnUpdateQuantityItem = jest.fn();

  const defaultPropsWithoutAction = {
    id: '1',
    image: 'https://example.com/image.jpg',
    name: 'Test Product',
    quantity: 2,
    price: 100,
    discount: 10,
  };

  const defaultProps = {
    ...defaultPropsWithoutAction,
    onRemoveItem: mockOnRemoveItem,
    onUpdateQuantityItem: mockOnUpdateQuantityItem,
  };

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('Should renders CartItem correctly', () => {
    render(<CartItem {...defaultProps} />);

    expect(screen.getByText('Test Product')).toBeTruthy();
    expect(screen.getByText('$90')).toBeTruthy();
    expect(screen.getByText('$100')).toBeTruthy();
    expect(screen.getByText('10% off')).toBeTruthy();
    expect(screen.getByText('Remove')).toBeTruthy();
    expect(screen.toJSON()).toMatchSnapshot();
  });

  it('Should renders CartItem correctly without actions', () => {
    render(<CartItem {...defaultPropsWithoutAction} />);

    expect(screen.getByText('Test Product')).toBeTruthy();
    expect(screen.getByText('$90')).toBeTruthy();
    expect(screen.getByText('$100')).toBeTruthy();
    expect(screen.getByText('10% off')).toBeTruthy();
    expect(screen.toJSON()).toMatchSnapshot();
  });

  it('Should calls onRemoveItem when "Remove" button is pressed', () => {
    render(<CartItem {...defaultProps} />);

    fireEvent.press(screen.getByText('Remove'));

    expect(mockOnRemoveItem).toHaveBeenCalledTimes(1);
    expect(mockOnRemoveItem).toHaveBeenCalledWith('1');
  });

  it('Should does not display discount if there is no discount', () => {
    render(<CartItem {...defaultProps} discount={undefined} />);

    expect(screen.getByText('$100')).toBeTruthy();
    expect(screen.queryByText('10% off')).toBeFalsy();
  });
});
