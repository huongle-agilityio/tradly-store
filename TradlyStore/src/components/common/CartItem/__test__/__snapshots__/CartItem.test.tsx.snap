// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartItem Should renders CartItem correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "",
    }
  }
>
  <View
    style={
      [
        [
          {
            "flexDirection": "row",
            "paddingHorizontal": 16,
          },
          {
            "gap": 15,
            "paddingBottom": 12,
            "paddingTop": 30,
          },
        ],
      ]
    }
  >
    <Image
      accessibilityLabel="image of Test Product product"
      accessibilityRole="image"
      source={
        {
          "uri": "https://example.com/image.jpg",
        }
      }
      style={
        {
          "borderRadius": 10,
          "height": 120,
          "width": 120,
        }
      }
    />
    <View
      style={
        {
          "justifyContent": "space-between",
          "paddingBottom": 10,
          "paddingTop": 14,
        }
      }
    >
      <Text
        style={
          [
            {
              "fontSize": 14,
              "lineHeight": 16,
            },
            {
              "fontFamily": "Montserrat-Medium",
              "fontWeight": "500",
            },
            {
              "color": "",
            },
            undefined,
          ]
        }
      >
        Test Product
      </Text>
      <View
        style={
          {
            "flexDirection": "row",
            "gap": 14,
          }
        }
      >
        <Text
          style={
            [
              {
                "fontSize": 18,
                "lineHeight": 18,
              },
              {
                "fontFamily": "Montserrat-SemiBold",
                "fontWeight": "700",
              },
              {
                "color": "",
              },
              undefined,
            ]
          }
        >
          $
          90
        </Text>
        <View
          style={
            {
              "flexDirection": "row",
              "gap": 5,
            }
          }
        >
          <Text
            style={
              [
                {
                  "fontSize": 14,
                  "lineHeight": 16,
                },
                {
                  "fontFamily": "Montserrat-Medium",
                  "fontWeight": "500",
                },
                {
                  "color": "",
                },
                {
                  "textDecorationLine": "line-through",
                },
              ]
            }
          >
            $
            100
          </Text>
          <Text
            style={
              [
                {
                  "fontSize": 14,
                  "lineHeight": 16,
                },
                {
                  "fontFamily": "Montserrat-Medium",
                  "fontWeight": "500",
                },
                {
                  "color": "",
                },
                undefined,
              ]
            }
          >
            10
            % off
          </Text>
        </View>
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
          }
        }
      >
        <Text
          style={
            [
              {
                "fontSize": 12,
                "lineHeight": 21,
              },
              {
                "fontFamily": "Montserrat-Medium",
                "fontWeight": "500",
              },
              {
                "color": "",
              },
              undefined,
            ]
          }
        >
          Qty:
        </Text>
        <View
          style={
            {
              "width": 60,
            }
          }
        >
          <View
            accessibilityLabel="Dropdown button"
            accessibilityRole="button"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": false,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-between",
                "opacity": 1,
                "padding": 12,
              }
            }
            testID="single-select-modal"
          >
            <Text
              style={
                [
                  {
                    "fontSize": 12,
                    "lineHeight": 21,
                  },
                  {
                    "fontFamily": "Montserrat-Medium",
                    "fontWeight": "500",
                  },
                  {
                    "color": "",
                  },
                  undefined,
                ]
              }
            >
              2
            </Text>
            <RNSVGSvgViewAndroid
              align="xMidYMid"
              bbHeight={4}
              bbWidth={9}
              color="#4f4f4f"
              focusable={false}
              height={4}
              meetOrSlice={0}
              minX={0}
              minY={0}
              style={
                [
                  {
                    "backgroundColor": "transparent",
                    "borderWidth": 0,
                  },
                  {
                    "flex": 0,
                    "height": 4,
                    "width": 9,
                  },
                ]
              }
              vbHeight={4}
              vbWidth={9}
              width={9}
            >
              <RNSVGGroup
                fill={
                  {
                    "payload": -16777216,
                    "type": 0,
                  }
                }
              >
                <RNSVGPath
                  clipRule={0}
                  d="M8.47396 0C8.65786 0 8.74428 0.227303 8.60683 0.349482L4.63287 3.88189C4.5571 3.94925 4.4429 3.94925 4.36713 3.88189L0.393167 0.349482C0.255716 0.227303 0.342136 0 0.52604 0H8.47396Z"
                  fill={
                    {
                      "type": 2,
                    }
                  }
                  fillRule={0}
                  propList={
                    [
                      "fill",
                      "fillRule",
                    ]
                  }
                />
              </RNSVGGroup>
            </RNSVGSvgViewAndroid>
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    accessibilityRole="button"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "borderTopColor": "",
        "borderTopWidth": 0.5,
        "opacity": 1,
        "paddingVertical": 12,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        [
          {
            "fontSize": 14,
            "lineHeight": 16,
          },
          {
            "fontFamily": "Montserrat-Medium",
            "fontWeight": "500",
          },
          {
            "color": "",
          },
          undefined,
        ]
      }
    >
      Remove
    </Text>
  </View>
</View>
`;

exports[`CartItem Should renders CartItem correctly without actions 1`] = `
<View
  style={
    {
      "backgroundColor": "",
    }
  }
>
  <View
    style={
      [
        [
          {
            "flexDirection": "row",
            "paddingHorizontal": 16,
          },
          {
            "gap": 15,
            "paddingVertical": 25,
          },
        ],
      ]
    }
  >
    <Image
      accessibilityLabel="image of Test Product product"
      accessibilityRole="image"
      source={
        {
          "uri": "https://example.com/image.jpg",
        }
      }
      style={
        {
          "borderRadius": 10,
          "height": 120,
          "width": 120,
        }
      }
    />
    <View
      style={
        {
          "justifyContent": "space-between",
          "paddingBottom": 10,
          "paddingTop": 14,
        }
      }
    >
      <Text
        style={
          [
            {
              "fontSize": 14,
              "lineHeight": 16,
            },
            {
              "fontFamily": "Montserrat-Medium",
              "fontWeight": "500",
            },
            {
              "color": "",
            },
            undefined,
          ]
        }
      >
        Test Product
      </Text>
      <View
        style={
          {
            "flexDirection": "row",
            "gap": 14,
          }
        }
      >
        <Text
          style={
            [
              {
                "fontSize": 18,
                "lineHeight": 18,
              },
              {
                "fontFamily": "Montserrat-SemiBold",
                "fontWeight": "700",
              },
              {
                "color": "",
              },
              undefined,
            ]
          }
        >
          $
          90
        </Text>
        <View
          style={
            {
              "flexDirection": "row",
              "gap": 5,
            }
          }
        >
          <Text
            style={
              [
                {
                  "fontSize": 14,
                  "lineHeight": 16,
                },
                {
                  "fontFamily": "Montserrat-Medium",
                  "fontWeight": "500",
                },
                {
                  "color": "",
                },
                {
                  "textDecorationLine": "line-through",
                },
              ]
            }
          >
            $
            100
          </Text>
          <Text
            style={
              [
                {
                  "fontSize": 14,
                  "lineHeight": 16,
                },
                {
                  "fontFamily": "Montserrat-Medium",
                  "fontWeight": "500",
                },
                {
                  "color": "",
                },
                undefined,
              ]
            }
          >
            10
            % off
          </Text>
        </View>
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
          }
        }
      >
        <Text
          style={
            [
              {
                "fontSize": 12,
                "lineHeight": 21,
              },
              {
                "fontFamily": "Montserrat-Medium",
                "fontWeight": "500",
              },
              {
                "color": "",
              },
              undefined,
            ]
          }
        >
          Qty:
        </Text>
        <View
          style={
            {
              "width": 60,
            }
          }
        >
          <View
            accessibilityLabel="Dropdown button"
            accessibilityRole="button"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": true,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={false}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-between",
                "opacity": 1,
                "padding": 12,
              }
            }
            testID="single-select-modal"
          >
            <Text
              style={
                [
                  {
                    "fontSize": 12,
                    "lineHeight": 21,
                  },
                  {
                    "fontFamily": "Montserrat-Medium",
                    "fontWeight": "500",
                  },
                  {
                    "color": "",
                  },
                  undefined,
                ]
              }
            >
              2
            </Text>
            <RNSVGSvgViewAndroid
              align="xMidYMid"
              bbHeight={4}
              bbWidth={9}
              color="#4f4f4f"
              focusable={false}
              height={4}
              meetOrSlice={0}
              minX={0}
              minY={0}
              style={
                [
                  {
                    "backgroundColor": "transparent",
                    "borderWidth": 0,
                  },
                  {
                    "flex": 0,
                    "height": 4,
                    "width": 9,
                  },
                ]
              }
              vbHeight={4}
              vbWidth={9}
              width={9}
            >
              <RNSVGGroup
                fill={
                  {
                    "payload": -16777216,
                    "type": 0,
                  }
                }
              >
                <RNSVGPath
                  clipRule={0}
                  d="M8.47396 0C8.65786 0 8.74428 0.227303 8.60683 0.349482L4.63287 3.88189C4.5571 3.94925 4.4429 3.94925 4.36713 3.88189L0.393167 0.349482C0.255716 0.227303 0.342136 0 0.52604 0H8.47396Z"
                  fill={
                    {
                      "type": 2,
                    }
                  }
                  fillRule={0}
                  propList={
                    [
                      "fill",
                      "fillRule",
                    ]
                  }
                />
              </RNSVGGroup>
            </RNSVGSvgViewAndroid>
          </View>
          <View
            backdropComponent={[Function]}
            enableDynamicSizing={false}
            enablePanDownToClose={true}
            index={-1}
            snapPoints={
              [
                "25%",
                "50%",
              ]
            }
            testID="bottom-sheet"
          >
            <RCTScrollView
              contentContainerStyle={
                {
                  "paddingHorizontal": 20,
                  "paddingVertical": 20,
                }
              }
              data={
                [
                  {
                    "label": "1",
                    "value": "1",
                  },
                  {
                    "label": "2",
                    "value": "2",
                  },
                  {
                    "label": "3",
                    "value": "3",
                  },
                  {
                    "label": "4",
                    "value": "4",
                  },
                  {
                    "label": "5",
                    "value": "5",
                  },
                  {
                    "label": "6",
                    "value": "6",
                  },
                  {
                    "label": "7",
                    "value": "7",
                  },
                  {
                    "label": "8",
                    "value": "8",
                  },
                  {
                    "label": "9",
                    "value": "9",
                  },
                  {
                    "label": "10",
                    "value": "10",
                  },
                ]
              }
              getItem={[Function]}
              getItemCount={[Function]}
              keyExtractor={[Function]}
              onContentSizeChange={[Function]}
              onLayout={[Function]}
              onMomentumScrollBegin={[Function]}
              onMomentumScrollEnd={[Function]}
              onScroll={[Function]}
              onScrollBeginDrag={[Function]}
              onScrollEndDrag={[Function]}
              removeClippedSubviews={true}
              renderItem={[Function]}
              scrollEventThrottle={0.0001}
              stickyHeaderIndices={[]}
              testID="bottom-sheet-flatlist"
              viewabilityConfigCallbackPairs={[]}
            >
              <View>
                <View
                  onFocusCapture={[Function]}
                  onLayout={[Function]}
                  style={null}
                >
                  <View
                    accessibilityRole="button"
                    accessibilityState={
                      {
                        "busy": undefined,
                        "checked": undefined,
                        "disabled": undefined,
                        "expanded": undefined,
                        "selected": undefined,
                      }
                    }
                    accessibilityValue={
                      {
                        "max": undefined,
                        "min": undefined,
                        "now": undefined,
                        "text": undefined,
                      }
                    }
                    accessible={true}
                    collapsable={false}
                    focusable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      {
                        "borderBottomColor": "",
                        "borderBottomWidth": 1,
                        "flexDirection": "row",
                        "justifyContent": "space-between",
                        "opacity": 1,
                        "paddingVertical": 10,
                      }
                    }
                    testID="option-item"
                  >
                    <Text
                      style={
                        {
                          "fontSize": 16,
                        }
                      }
                    >
                      1
                    </Text>
                  </View>
                </View>
                <View
                  onFocusCapture={[Function]}
                  onLayout={[Function]}
                  style={null}
                >
                  <View
                    accessibilityRole="button"
                    accessibilityState={
                      {
                        "busy": undefined,
                        "checked": undefined,
                        "disabled": undefined,
                        "expanded": undefined,
                        "selected": undefined,
                      }
                    }
                    accessibilityValue={
                      {
                        "max": undefined,
                        "min": undefined,
                        "now": undefined,
                        "text": undefined,
                      }
                    }
                    accessible={true}
                    collapsable={false}
                    focusable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      {
                        "borderBottomColor": "",
                        "borderBottomWidth": 1,
                        "flexDirection": "row",
                        "justifyContent": "space-between",
                        "opacity": 1,
                        "paddingVertical": 10,
                      }
                    }
                    testID="option-item"
                  >
                    <Text
                      style={
                        {
                          "fontSize": 16,
                        }
                      }
                    >
                      2
                    </Text>
                    <Text
                      style={
                        [
                          {
                            "fontSize": 16,
                          },
                          {
                            "color": "",
                          },
                        ]
                      }
                    >
                      ✔
                    </Text>
                  </View>
                </View>
                <View
                  onFocusCapture={[Function]}
                  onLayout={[Function]}
                  style={null}
                >
                  <View
                    accessibilityRole="button"
                    accessibilityState={
                      {
                        "busy": undefined,
                        "checked": undefined,
                        "disabled": undefined,
                        "expanded": undefined,
                        "selected": undefined,
                      }
                    }
                    accessibilityValue={
                      {
                        "max": undefined,
                        "min": undefined,
                        "now": undefined,
                        "text": undefined,
                      }
                    }
                    accessible={true}
                    collapsable={false}
                    focusable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      {
                        "borderBottomColor": "",
                        "borderBottomWidth": 1,
                        "flexDirection": "row",
                        "justifyContent": "space-between",
                        "opacity": 1,
                        "paddingVertical": 10,
                      }
                    }
                    testID="option-item"
                  >
                    <Text
                      style={
                        {
                          "fontSize": 16,
                        }
                      }
                    >
                      3
                    </Text>
                  </View>
                </View>
                <View
                  onFocusCapture={[Function]}
                  onLayout={[Function]}
                  style={null}
                >
                  <View
                    accessibilityRole="button"
                    accessibilityState={
                      {
                        "busy": undefined,
                        "checked": undefined,
                        "disabled": undefined,
                        "expanded": undefined,
                        "selected": undefined,
                      }
                    }
                    accessibilityValue={
                      {
                        "max": undefined,
                        "min": undefined,
                        "now": undefined,
                        "text": undefined,
                      }
                    }
                    accessible={true}
                    collapsable={false}
                    focusable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      {
                        "borderBottomColor": "",
                        "borderBottomWidth": 1,
                        "flexDirection": "row",
                        "justifyContent": "space-between",
                        "opacity": 1,
                        "paddingVertical": 10,
                      }
                    }
                    testID="option-item"
                  >
                    <Text
                      style={
                        {
                          "fontSize": 16,
                        }
                      }
                    >
                      4
                    </Text>
                  </View>
                </View>
                <View
                  onFocusCapture={[Function]}
                  onLayout={[Function]}
                  style={null}
                >
                  <View
                    accessibilityRole="button"
                    accessibilityState={
                      {
                        "busy": undefined,
                        "checked": undefined,
                        "disabled": undefined,
                        "expanded": undefined,
                        "selected": undefined,
                      }
                    }
                    accessibilityValue={
                      {
                        "max": undefined,
                        "min": undefined,
                        "now": undefined,
                        "text": undefined,
                      }
                    }
                    accessible={true}
                    collapsable={false}
                    focusable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      {
                        "borderBottomColor": "",
                        "borderBottomWidth": 1,
                        "flexDirection": "row",
                        "justifyContent": "space-between",
                        "opacity": 1,
                        "paddingVertical": 10,
                      }
                    }
                    testID="option-item"
                  >
                    <Text
                      style={
                        {
                          "fontSize": 16,
                        }
                      }
                    >
                      5
                    </Text>
                  </View>
                </View>
                <View
                  onFocusCapture={[Function]}
                  onLayout={[Function]}
                  style={null}
                >
                  <View
                    accessibilityRole="button"
                    accessibilityState={
                      {
                        "busy": undefined,
                        "checked": undefined,
                        "disabled": undefined,
                        "expanded": undefined,
                        "selected": undefined,
                      }
                    }
                    accessibilityValue={
                      {
                        "max": undefined,
                        "min": undefined,
                        "now": undefined,
                        "text": undefined,
                      }
                    }
                    accessible={true}
                    collapsable={false}
                    focusable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      {
                        "borderBottomColor": "",
                        "borderBottomWidth": 1,
                        "flexDirection": "row",
                        "justifyContent": "space-between",
                        "opacity": 1,
                        "paddingVertical": 10,
                      }
                    }
                    testID="option-item"
                  >
                    <Text
                      style={
                        {
                          "fontSize": 16,
                        }
                      }
                    >
                      6
                    </Text>
                  </View>
                </View>
                <View
                  onFocusCapture={[Function]}
                  onLayout={[Function]}
                  style={null}
                >
                  <View
                    accessibilityRole="button"
                    accessibilityState={
                      {
                        "busy": undefined,
                        "checked": undefined,
                        "disabled": undefined,
                        "expanded": undefined,
                        "selected": undefined,
                      }
                    }
                    accessibilityValue={
                      {
                        "max": undefined,
                        "min": undefined,
                        "now": undefined,
                        "text": undefined,
                      }
                    }
                    accessible={true}
                    collapsable={false}
                    focusable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      {
                        "borderBottomColor": "",
                        "borderBottomWidth": 1,
                        "flexDirection": "row",
                        "justifyContent": "space-between",
                        "opacity": 1,
                        "paddingVertical": 10,
                      }
                    }
                    testID="option-item"
                  >
                    <Text
                      style={
                        {
                          "fontSize": 16,
                        }
                      }
                    >
                      7
                    </Text>
                  </View>
                </View>
                <View
                  onFocusCapture={[Function]}
                  onLayout={[Function]}
                  style={null}
                >
                  <View
                    accessibilityRole="button"
                    accessibilityState={
                      {
                        "busy": undefined,
                        "checked": undefined,
                        "disabled": undefined,
                        "expanded": undefined,
                        "selected": undefined,
                      }
                    }
                    accessibilityValue={
                      {
                        "max": undefined,
                        "min": undefined,
                        "now": undefined,
                        "text": undefined,
                      }
                    }
                    accessible={true}
                    collapsable={false}
                    focusable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      {
                        "borderBottomColor": "",
                        "borderBottomWidth": 1,
                        "flexDirection": "row",
                        "justifyContent": "space-between",
                        "opacity": 1,
                        "paddingVertical": 10,
                      }
                    }
                    testID="option-item"
                  >
                    <Text
                      style={
                        {
                          "fontSize": 16,
                        }
                      }
                    >
                      8
                    </Text>
                  </View>
                </View>
                <View
                  onFocusCapture={[Function]}
                  onLayout={[Function]}
                  style={null}
                >
                  <View
                    accessibilityRole="button"
                    accessibilityState={
                      {
                        "busy": undefined,
                        "checked": undefined,
                        "disabled": undefined,
                        "expanded": undefined,
                        "selected": undefined,
                      }
                    }
                    accessibilityValue={
                      {
                        "max": undefined,
                        "min": undefined,
                        "now": undefined,
                        "text": undefined,
                      }
                    }
                    accessible={true}
                    collapsable={false}
                    focusable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      {
                        "borderBottomColor": "",
                        "borderBottomWidth": 1,
                        "flexDirection": "row",
                        "justifyContent": "space-between",
                        "opacity": 1,
                        "paddingVertical": 10,
                      }
                    }
                    testID="option-item"
                  >
                    <Text
                      style={
                        {
                          "fontSize": 16,
                        }
                      }
                    >
                      9
                    </Text>
                  </View>
                </View>
                <View
                  onFocusCapture={[Function]}
                  onLayout={[Function]}
                  style={null}
                >
                  <View
                    accessibilityRole="button"
                    accessibilityState={
                      {
                        "busy": undefined,
                        "checked": undefined,
                        "disabled": undefined,
                        "expanded": undefined,
                        "selected": undefined,
                      }
                    }
                    accessibilityValue={
                      {
                        "max": undefined,
                        "min": undefined,
                        "now": undefined,
                        "text": undefined,
                      }
                    }
                    accessible={true}
                    collapsable={false}
                    focusable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      {
                        "flexDirection": "row",
                        "justifyContent": "space-between",
                        "opacity": 1,
                        "paddingVertical": 10,
                      }
                    }
                    testID="option-item"
                  >
                    <Text
                      style={
                        {
                          "fontSize": 16,
                        }
                      }
                    >
                      10
                    </Text>
                  </View>
                </View>
              </View>
            </RCTScrollView>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
